src/validation/mod.rs

```diff
@@ -162,94 +164,96 @@ pub fn validate(wasm: &[u8]) -> Result<ValidationInfo> {
-    //         //     // passive or declarative
-    //         //     if ttype & 0b010 == 0b010 {
-    //         //         ElemMode::Declarative
-    //         //     } else {
-    //         //         ElemMode::Passive
-    //         //     }
-    //         // } else {
-    //         //     if ttype & 0b010 == 0b010 {
-    //         //         let table_idx = wasm.read_var_u32().unwrap();
-    //         //         let bytes = Vec::new();
-    //         //         bytes.push(wasm.read_var_u32().unwrap());
-    //         //         while bytes.last().unwrap() != END {
-    //         //             bytes.push(wasm.read_var_u32().unwrap());
-    //         //         }
-    //         //         ElemMode::Active(ActiveElem {
-    //         //             table: table_idx,
-    //         //             offset: bytes
-    //         //         })
-    //         //     } else {
-    //         //     }
-    //         // }
-    //         match ttype {
-    //             0 => {
-    //                 let expr = {
-    //                     // TODO: actually verify this expression
-    //                     let mut const_expr = read_constant_expression(wasm).unwrap();
-    //                 };
-    //                 let func_idxs: Vec<u32> = wasm.read_vec(|w| {
-    //                     w.read_var_u32()
-    //                 }).unwrap();
-    //                 // type funcref
-    //             }
-    //             1 => {
-    //                 // type elemkind
-    //             }
-    //             2 => {
-    //                 // type elemkind
-    //             }
-    //             3 => {
-    //                 // type elemkind
-    //             }
-    //             4 => {
-    //                 // type funcref
-    //             }
-    //             5 => {
-    //                 // type reftype
-    //             }
-    //             6 => {
-    //                 // type reftype
-    //             }
-    //             7 => {
-    //                 // type reftype
-    //             }
-    //             _ => unimplemented!()
-    //         }
-    //      }
-    //     todo!("element section not yet supported")
-    // })?;
-    // #endregion
-
-    let _element: Option<()> =
-        handle_section(&mut wasm, &mut header, SectionTy::Element, |_, _| {
-            todo!("element section not yet supported")
-        })?;
-    while (skip_section(&mut wasm, &mut header)?).is_some() {}
+    let elements: Vec<ElemType> = handle_section(&mut wasm, &mut header, SectionTy::Element, |wasm, _| {
+        use crate::core::reader::types::element::*;
+        use crate::RefType;
+        let mut elem_vec: Vec<ElemType> = Vec::new();
+        // https://webassembly.github.io/spec/core/binary/modules.html#element-section
+        // TODO: replace with wasm.read_vec in the future
+        let vec_length = wasm.read_var_u32().unwrap();
+        trace!("Element sections no.: {}", vec_length);
```

Comment:
Does this mean to imply there can be multiple element sections in the wasm binary? I'm guessing this is a typo?

```suggestion
        trace!("Validating element sections with {} elements", vec_length);
```